File: engine/demo/src/main/scala/pl/touk/nussknacker/engine/demo/UtilProcessHelper.scala
package pl.touk.nussknacker.engine.demo

import io.circe.Encoder
import pl.touk.nussknacker.engine.api.{Documentation, HideToString, ParamName}

import scala.util.Random

object UtilProcessHelper extends HideToString {

  import scala.collection.JavaConverters._

  @Documentation(description = "Convert map to JSON")
  def mapToJson(map: java.util.Map[String, String]): String = {
    Encoder.encodeMap[String, String].apply(map.asScala.toMap).noSpaces
  }

  @Documentation(description = "Get random number")
  def random(@ParamName("To (exclusive)") to: Int): Unit = {
    Random.nextInt(to)

  }
}

File: engine/demo/src/main/java/pl/touk/nussknacker/engine/javademo/UtilProcessHelper.java
package pl.touk.nussknacker.engine.javademo;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import pl.touk.nussknacker.engine.api.HideToString;

public class UtilProcessHelper implements HideToString {

    public String mapToJson(java.util.Map<String, String> map) throws IOException {
        return new ObjectMapper().writeValueAsString(map);
    }
}
